python/smqtk/detect_and_describe/colordescriptor/color_detector_and_descriptor.py

```diff
@@ -0,0 +1,633 @@
+                    r_map[uid] = None
+                    continue
+                finally:
+                    # Done with image file, so remove from filesystem
+                    tmp_clean_method()
+
+                if d_width is None and d_shape[0] != 0:
+                    f_width = f_shape[1]
+                    d_width = d_shape[1]
+
+                # skip this result if it generated no descriptors
+                if d_shape[1] == 0:
+                    continue
+
+                ssi = None
+                if f_shape[0] > per_item_limit:
+                    # pick random indices to subsample down to size limit
+                    ssi = sorted(
+                        numpy.random.permutation(f_shape[0])[:per_item_limit]
+                    )
+
+                # Only keep this if any descriptors were generated
+                r_map[uid] = (ffp, dfp, running_height, ssi)
+                running_height += min(f_shape[0], per_item_limit)
+            pool.join()
+
+            # Asynchronously load files, inserting data into master matrices
+            self._log.debug("Building super matrices...")
+            master_feat = numpy.zeros((running_height, f_width), dtype=float)
+            master_desc = numpy.zeros((running_height, d_width), dtype=float)
+            tp = multiprocessing.pool.ThreadPool(processes=self.PARALLEL)
+            for uid in s_keys:
+                if r_map[uid]:
+                    ffp, dfp, sR, ssi = r_map[uid]
+                    tp.apply_async(ColorDetectAndDescribe_Image._thread_load_matrix,
+                                   args=(ffp, master_feat, sR, ssi))
+                    tp.apply_async(ColorDetectAndDescribe_Image._thread_load_matrix,
+                                   args=(dfp, master_desc, sR, ssi))
+            tp.close()
+            tp.join()
+
+            # We want to convert the matrix to a DescriptorElement.
+            descriptor_element, _ = npy_array_to_descriptor_element_md5(self._data_element_type, 
+                self._descriptor_element_factory, 
+                self.name, master_desc)
+
+            print descriptor_element
+
+            return master_feat, descriptor_element
+
+    def valid_content_types(self):
+        """
+        :return: A set valid MIME type content types that this descriptor can
+            handle.
+        :rtype: set[str]
+        """
+        return {'image/bmp', 'image/jpeg', 'image/png', 'image/tiff'}
+
+    @staticmethod
+    def _thread_load_matrix(filepath, m, sR, subsample=None):
+        """
+        load a numpy matrix from ``filepath``, inserting the loaded matrix into
+        ``m`` starting at the row ``sR``.
+
+        If subsample has a value, it will be a list if indices to
+        """
+        n = numpy.load(filepath)
+        if subsample:
+            n = n[subsample, :]
+        m[sR:sR+n.shape[0], :n.shape[1]] = n    
+
+
+class ColorDetectAndDescribe_Video (ColorDetectAndDescribe_Base):
```

Comment:
This class looks to be inoperable in the current state as required imports are missing and variable names need to be changed. Changes look trivial but would need testing.